api-example.xml

```diff
@@ -0,0 +1,129 @@
+      gravity = answer['vectors']['g']
+      if not compare_lengths(normal, gravity):
+          return 'Normal Force and Gravitational Force should be similar lengths.'
+
+  def check_normal_angle(answer):
+      normal = answer['vectors']['N']
+      if not check_angle(normal, 90, tolerance=2.0):
+          return 'The angle of the Normal Force is incorrect. Your angle: %.1f' % normal.angle
+
+  def check_gravity_angle(answer):
+      gravity = answer['vectors']['g']
+      if not check_angle(gravity, 270.0, tolerance=2.0):
+          return 'The angle of the Gravitational Force is incorrect. Your angle: %.1f' % gravity.angle
+
+  def check_friction_angle(answer):
+      friction = answer['vectors']['f']
+      if not check_angle(friction, 180.0, tolerance=2.0):
+          return 'The angle of the Frictional Force is incorrect. Your angle: %.1f' % friction.angle
+
+  def check_starts_at_center_of_mass(answer, tolerance=0.25):
+      cm = answer['points']['cm']
+      for vec in answer['vectors'].values():
+          dist = math.hypot(cm.x - vec.tail.x, cm.y - vec.tail.y)
+          if dist > tolerance:
+              return 'One or more of your vectors do not start from the center of mass.'
+
+  checks = [
+      check_all_vectors_present,
+      check_compare_normal_and_gravity_lengths,
+      check_normal_angle,
+      check_gravity_angle,
+      check_friction_angle,
+      check_starts_at_center_of_mass,
+  ]
+  success_message = 'Good job!'
+
+
+  ### Python API - Do not edit after this line ####
+
+  class Point(object):
+      def __init__(self, x, y):
+          self.x = x
+          self.y = y
+
+  class Vector(object):
+      def __init__(self, x1, y1, x2, y2):
+          self.tail = Point(x1, y1)
+          self.tip = Point(x2, y2)
+          self.length = math.hypot(x2 - x1, y2 - y1)
+          angle = math.degrees(math.atan2(y2 - y1, x2 - x1))
+          if angle < 0:
+              angle += 360
+          self.angle = angle
+
+  class Grader(object):
+      def __init__(self, answer, checks, success_message='Test passed'):
+          self.answer = answer
+          self.checks = checks
+          self.success_message = success_message
+
+      def grade(self):
+          for check in self.checks:
+              result = check(self._convert_answer())
+              if result:
+                  return {'ok': False, 'msg': result}
+          return {'ok': True, 'msg': self.success_message}
+
+      def _convert_answer(self):
+          vectors = {}
+          for name, props in self.answer['vectors'].iteritems():
+              tail = props['tail']
+              tip = props['tip']
+              vectors[name] = Vector(tail[0], tail[1], tip[0], tip[1])
+
+          points = {}
+          for name, props in self.answer['points'].iteritems():
+              coords = props['coords']
+              points[name] = Point(coords[0], coords[1])
+
+          return {'vectors': vectors, 'points': points}
```

Comment:
@mtyaka nit: in my opinion, using tuple here and changing `check` signature to `def check(vectors, points)` would yield a cleaner checks, e.g.: `friction = answer['vectors']['f']` would be `friction = vectors['f']`. This differs a bit from what I said in task discussion and have its disadvantages, so if you think this is good enough already - left it as is.